lib/http_server/response/response.ex

```diff
@@ -24,4 +24,10 @@ defmodule HttpServer.Response.Response do
       404 => "Not Found"
     }[code]
   end
+
+  def update_headers(response) when response.body != nil do
```

Comment:
If we are expecting a %Response{} struct to be passed in, you can make sure that the function pattern matches on it, to prevent incorrect data being passed in. (you probably want to do this anytime in the code you are passing around a custom Struct.

```suggestion
  def update_headers(%Response{} = response) when response.body != nil do
```

Then, anytime `#update_headers/1` is called with something other than a response you could ignore it, or maybe even raise an error.

```exlixir
  def update_headers(_), do: nil
```

Reply:
This is a good suggestion. I'll change it and will keep it in mind for the future.